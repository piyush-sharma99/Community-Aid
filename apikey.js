/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
