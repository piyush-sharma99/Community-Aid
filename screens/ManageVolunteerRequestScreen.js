/*
 *  ClassName: ManageVolunteerRequestScreen.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=iMCM1NceGJY
 */

//Imports
import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  TextInput,
  TouchableOpacity,
  FlatList,
  SafeAreaView,
} from "react-native";
import { Card } from "react-native-paper";
import * as firebase from "firebase";
import editStatus from "../functions/editStatus";
import unassign from "../functions/unassign";
import completeRequest from "../functions/completeRequest";

//Main Component
const ManageVolunteerRequestScreen = (props) => {
  //Initialising Variables
  const [requests, setRequests] = useState([]);
  const db = firebase.firestore();
  const [request, setRequest] = useState("");
  const [status, setStatus] = useState("");
  var user = firebase.auth().currentUser;

  //The function below reads assistance requests available for picked up by the user
  useEffect(() => {
    db.collection("Assistance Request")
      .where("vid", "==", user.uid)
      .onSnapshot((querySnapshot) => {
        var requests = [];

        querySnapshot.forEach((documentSnapshot) => {
          requests.push({
            ...documentSnapshot.data(),
            key: documentSnapshot.request_ID,
          });
        });

        requests = requests.sort(
          (a, b) => b.Time["seconds"] - a.Time["seconds"]
        );
        setRequests(requests);
        console.log(requests);
      });
  }, []);

  //Function that empties fields
  resetFieldsOne = () => {
    this.textInputTwo.clear();
  };

  //The function below takes data previously read above and displays it using the structure below in a flatlist format
  const renderRequest = ({ item }) => (
    <View style={styles.screen2}>
      <Card style={styles.cardView2}>
        <View>
          <Text style={styles.text2}>Request Time/Date:</Text>
          <Text style={styles.text3}>{item.date}</Text>
          <Text style={styles.text2}>Request Type:</Text>
          <Text style={styles.text3}>{item.request_Type}</Text>
          <Text style={styles.text2}>Request Description:</Text>
          <Text style={styles.text3}>{item.request_Description}</Text>
          <Text style={styles.text2}>Request Status:</Text>
          <Text style={styles.text3}>{item.status}</Text>
        </View>

        <View style={styles.inputView}>
          <TextInput
            style={styles.inputText}
            placeholder="Edit Status..."
            placeholderTextColor="#003f5c"
            onChangeText={(status) => setStatus(status)}
            ref={(input) => {
              this.textInputTwo = input;
            }}
          />
        </View>

        <View>
          <TouchableOpacity
            style={styles.Btn}
            onPressIn={() => editStatus(db, item, status)}
            onPress={resetFieldsOne}
          >
            <Text style={styles.Text}>Edit Request</Text>
          </TouchableOpacity>
        </View>
        <View>
          <TouchableOpacity
            style={styles.Btn}
            onPressIn={() => unassign(db, item)}
          >
            <Text style={styles.Text}>Cancel Request</Text>
          </TouchableOpacity>
        </View>
        <View>
          <TouchableOpacity
            style={styles.Btn}
            onPressIn={() => completeRequest(db, item)}
          >
            <Text style={styles.Text}>Completed</Text>
          </TouchableOpacity>
        </View>
      </Card>
    </View>
  );

  return (
    <ImageBackground source={require("../assets/BG.png")} style={styles.bImage}>
      <View style={styles.screen}>
        <Text style={styles.text}>Update Requests below:</Text>

        <FlatList
          //Flatlist for reading and displaying data
          style={{ flex: 1 }}
          data={requests}
          renderItem={renderRequest}
          keyExtractor={(item) => item.request_ID}
        />
      </View>
    </ImageBackground>
  );
};

//Navigation options: changing header displayed on the page
ManageVolunteerRequestScreen.navigationOptions = {
  headerTitle: "Manage Volunteer Requests",
  headerTitleStyle: { alignSelf: "center" },
  headerStyle: {
    backgroundColor: "#2c8ffa",
  },
  headerTintColor: "white",
};

//CSS
const styles = StyleSheet.create({
  screen: {
    width: "100%",
    height: "100%",
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  bImage: {
    width: "100%",
    height: "100%",
  },
  screen2: {
    width: 410,
    alignItems: "center",
    justifyContent: "center",
  },
  text: {
    padding: 10,
    marginBottom: 20,
    marginRight: 40,
    marginLeft: 40,
    color: "white",
    fontWeight: "bold",
    fontSize: 20,
    textAlign: "center",
    marginTop: 15,
    borderWidth: 3,
    borderColor: "#fb5b5a",
    borderRadius: 15,
  },
  text2: {
    padding: 5,
    marginRight: 40,
    marginLeft: 40,
    color: "white",
    fontWeight: "bold",
    fontSize: 20,
    textAlign: "center",
  },

  text3: {
    padding: 5,
    marginRight: 40,
    marginLeft: 40,
    color: "white",
    fontSize: 20,
    textAlign: "center",
  },

  inputView: {
    padding: 20,
    backgroundColor: "#FDFEFE",
    borderRadius: 25,
    height: 35,
    justifyContent: "center",
    elevation: 10,
    marginTop: 15,
    borderWidth: 4,
    borderColor: "#2E86C1",
  },

  inputText: {
    height: 50,
    color: "black",
  },

  Text: {
    color: "white",
    fontSize: 20,
  },

  Btn: {
    width: "100%",
    backgroundColor: "#2E86C1",
    borderRadius: 10,
    height: 40,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 15,
    borderWidth: 2,
    borderColor: "white",
  },

  cardView2: {
    justifyContent: "center",
    width: "95%",
    backgroundColor: "#fb5b5a",
    shadowColor: "black",
    borderRadius: 25,
    borderWidth: 3,
    borderColor: "#fff",
    shadowOpacity: 1,
    marginTop: 15,
    marginBottom: 20,
    padding: 15,
    elevation: 10,
    alignItems: "center",
    shadowOffset: {
      width: 3,
      height: 3,
    },
  },
});

export default ManageVolunteerRequestScreen;
