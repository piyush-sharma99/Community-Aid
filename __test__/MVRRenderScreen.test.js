import React from "react";
import ManageVolunteerRequestScreen from "../screens/ManageVolunteerRequestScreen";
import { firebaseConfig } from "../apikey";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

test("ManageVolunteerRequest renders correctly", () => {
  const tree = renderer.create(<ManageVolunteerRequestScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
