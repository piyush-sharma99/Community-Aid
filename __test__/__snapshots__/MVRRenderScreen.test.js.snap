// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageVolunteerRequest renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fb5b5a",
          "borderRadius": 15,
          "borderWidth": 3,
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Update Requests below:
    </Text>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flex": 1,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
